common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -1267,6 +1268,119 @@ public static UTF8String[] icuSplitSQL(final UTF8String string, final UTF8String
     return strings.toArray(new UTF8String[0]);
   }
 
+  /**
+   * Title casing a string according to a new behaviour.
+   * Iterates over the string and title cases the first character in each word, and lowercases
+   * every other character.
+   * Handles lowercasing greek sigma(Σ) separately, taking into account if
+   * it should be a small final sigma(ς) or small non-final sigma(σ).
+   * Words are separated by ASCII space(\u0020).
+   *
+   * @param target UTF8String to be title cased
+   * @return title cased target
+   */
+  public static UTF8String toTitleCaseICU(UTF8String target) {
+
+    Iterator<Integer> codepointIterator = target.
+            codePointIterator(CodePointIteratorType.CODE_POINT_ITERATOR_MAKE_VALID);
+
+    // building the title cased target with sb
+    StringBuilder sb=new StringBuilder();
```

Comment:
```suggestion
    StringBuilder sb = new StringBuilder();
```
please follow this code style guide:
https://github.com/databricks/scala-style-guide

Reply:
(to avoid duplicate comments on the PR - this goes for all instances below)